src/codegen/optimizer.lisp

```diff
@@ -351,43 +351,95 @@ requires direct constructor calls."
            (type (integer 0)    max-unroll)
            (type function       heuristic)
            (values node &optional))
-  (flet ((try-inline (node call-stack)
-           (declare (type (or node-application node-direct-application) node)
-                    (type parser:identifier-list                        call-stack)
-                    (values (or node-let node-application node-direct-application) &optional))
-           (or
-            (alexandria:when-let*
-                ((name (node-rator-name node))
-                 (code (tc:lookup-code env name :no-error t))
-                 (_    (and (node-abstraction-p code)
+  (labels ((inline-abstraction-from-application (node abstraction process-body)
+             "If NODE is an abstraction (f e1 e2 ... en) and ABSTRACTION is a
```

Comment:
Do we prefer "`` NODE ``" instead of "`` `node` ``"?

Reply:
In Lisp code I've preferred capitalized names. But I actually forgot what our convention was.

Lisp code docstrings aren't emitted to markdown anywhere.